Add 'rho fact list' command.

noahl opened the pull request:
'rho list-facts' now shows documentation about all of the facts that
Rho can collect.

The documentation is kept in `rho/fact_docs.py`. This change also adds
a script at `docs/generate_python_docs.py` which generates
`fact_docs.py` from the reStructured Text documentation at
`doc/command_syntax_usage.rst`, to make it easy to keep them in sync.

Closes #80.

Commit:
Add 'rho list-facts' command.

'rho list-facts' now shows documentation about all of the facts that
Rho can collect.

The documentation is kept in `rho/fact_docs.py`. This change also adds
a script at `docs/generate_python_docs.py` which generates
`fact_docs.py` from the reStructured Text documentation at
`doc/command_syntax_usage.rst`, to make it easy to keep them in sync.

chambridge commented on doc/command_syntax_usage.rst:
```diff
@@ -14,6 +14,7 @@ There are eight rho commands:
  * ``edit`` - to modify an existing entry
  * ``clear`` - to remove any or all entries
  * ``show`` and ``list`` - to display one or more entries
+ * ``list-facts`` - to list the facts Rho can collect
```
I would prefer the command be named `rho fact list` which would fit the `rho <noun> <verb>` flow found in other commands.

noahl replied:
Renamed.

chambridge replied:
shouldn't the above documentation be updated?

chambridge reviewed (commented):
Build appears to be failing with an import error. Looks like you may need to add a new dependency.

Commit:
Add missing dependency and rename command

'rho list-facts' becomes 'rho fact list' in response to review
comments.

Commit:
Update tests for 'fact list' rename

Commit:
Makefile experiment

Trying to understand what's going wrong on Travis.

Commit:
Separate 'gen-python-docs' from 'make docs'

This makes sense, as we don't need to regenerate Python docs just to
generate the HTML documentation, and should also fix Travis issues.

noahl commented:
@chambridge I fixed the requirement issue and some Makefile problems; I believe this is ready for review.

Commit:
Rename 'list-facts' to 'fact list'

noahl commented:
Thanks! Fixed that.

On Wed, Aug 9, 2017 at 1:21 PM, Chris Hambridge <notifications@github.com>
wrote:

> *@chambridge* commented on this pull request.
> ------------------------------
>
> In doc/command_syntax_usage.rst
> <https://github.com/quipucords/rho/pull/156#discussion_r132249410>:
>
> > @@ -14,6 +14,7 @@ There are eight rho commands:
>   * ``edit`` - to modify an existing entry
>   * ``clear`` - to remove any or all entries
>   * ``show`` and ``list`` - to display one or more entries
> + * ``list-facts`` - to list the facts Rho can collect
>
> shouldn't the above documentation be updated?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/quipucords/rho/pull/156#discussion_r132249410>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AABB8rw5CQJBJH_g4ukR0munW2ckFCwYks5sWeqjgaJpZM4OxPmj>
> .
>

chambridge commented on doc/command_syntax_usage.rst:
```diff
@@ -14,7 +14,7 @@ There are eight rho commands:
  * ``clear`` - to remove any or all entries
  * ``show`` and ``list`` - to display one or more entries
- * ``list-facts`` - to list the facts Rho can collect
+ * ``fact`` - to show information about the facts Rho can collect
```
Sorry for the nit pick, but we tend to refer to `rho` in all lowercase in the docs. Just want to be consistent.

noahl replied:
Fixed, thank you!

chambridge reviewed (approved):
Minor comment, otherwise looks good to me.

Commit:
Lowercase 'rho'

noahl commented:
Thanks for taking the time to make the docs consistent.

On Wed, Aug 9, 2017 at 1:27 PM, Chris Hambridge <notifications@github.com>
wrote:

> *@chambridge* approved this pull request.
>
> Minor comment, otherwise looks good to me.
> ------------------------------
>
> In doc/command_syntax_usage.rst
> <https://github.com/quipucords/rho/pull/156#discussion_r132250750>:
>
> > @@ -14,7 +14,7 @@ There are eight rho commands:
>   * ``edit`` - to modify an existing entry
>   * ``clear`` - to remove any or all entries
>   * ``show`` and ``list`` - to display one or more entries
> - * ``list-facts`` - to list the facts Rho can collect
> + * ``fact`` - to show information about the facts Rho can collect
>
> Sorry for the nit pick, but we tend to refer to rho in all lowercase in
> the docs. Just want to be consistent.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/quipucords/rho/pull/156#pullrequestreview-55311574>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABB8i3VjEacOvz1INF8gW5_NLTshEleks5sWevzgaJpZM4OxPmj>
> .
>
